shared/aggregation/maxcover.go

```diff
@@ -95,6 +95,82 @@ func (mc *MaxCoverProblem) Cover(k int, allowOverlaps bool) (*Aggregation, error
 	return solution, nil
 }
 
+// MaxCover finds the k-cover of Maximum Coverage problem.
```

Comment:
I'd love to see more commentary on how to these input/out parameters. It'll be nice for Godoc.

Reply:
Our previous implementation has lots of documentary comments, for e.g.
https://github.com/prysmaticlabs/prysm/blob/d9c451d547ace0e2a331eba553d818700de781d6/shared/aggregation/maxcover.go#L14-L28

This optimized implementation -- once we are sure it works w/o problems -- will outdate that original implementation (but not the comments), so when removing outdated code I will make sure that previous comments are included with new methods (otherwise we will be duplicating lots of things).